src/server/index.jsx

```diff
@@ -142,10 +143,25 @@ server
       const { service, isAmp } = getRouteProps(routes, url);
       const bbcOrigin = headers['bbc-origin'];
 
+      let dials = {};
+      try {
```

Comment:
can this try catch go inside getDials? best not the clutter up this file as theres a lot going on, and each thing can be self contained

Reply:
The purpose of this is to log any errors with fetching dials and I don't want to move logging to getDials. This pattern is already being used by loadInitialData.

Reply:
>I don't want to move logging to getDials
Why not? this is done in many places

Reply:
Because I want to keep consistency of how we handle data fetching. We don't use this pattern in the other data fetching functions. If you think this should be refactored as a whole, by all means raise an issue.

Reply:
> we don't use this pattern in other data fetching functions

I don't think we should follow that principle of 'it's not done this way anywhere else so we should stick to it' when it compromises readability, usability, and clean code. It would make sense to move the try catch to the `getDials` file in my opinion.

Reply:
Absolutely, if it compromises readability, usability and clean code. I don’t understand how this code style conflicts with that.

In my opinion, moving this inside of helper functions makes them significantly less reusable as it locks down the error handling implementation, which is inefficient due to the potential requirement of handling errors different on the server vs client.

Whilst this specifically will not be run client side at the moment, core pod are actively working on features toggling that will be client side (either with Dials or some other service). Consistency in how we handle data fetching across the application is not to be underestimated or devalued.

If you think that maybe refactoring the server so that each route handler is separated into their own modules, by all means knock yourself out, but that is out of the scope of this issue.

Reply:
>significantly less reusable
Not sure how this function can be reused, it makes an `fs` file lookup for a specific file, we should only ever call this once

Reply:
At the moment it does, we could decide to expose this on an endpoint and fetch via a HTTP request to allow us to fetch dials client side.

Reply:
Surely that will also need error handling and logging though?

Reply:
It would, although the logging strategy is likely to be different.

Reply:
There's limited use trying to predict possible future code, we won't know till we get there. This implementation works, is IMO readable and can easily be changed later if needed, so let's stick with it and worry about bigger problems.